extensions/GitExtension/FileExplorerGitIntegration/Models/GitExecute.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using FileExplorerGitIntegration.Helpers;
+using Microsoft.Windows.DevHome.SDK;
+using Serilog;
+
+namespace FileExplorerGitIntegration.Models;
+
+public class GitExecute
+{
+    public static GitCommandRunnerResultInfo ExecuteGitCommand(string gitApplication, string repositoryDirectory, string command, string arguments)
+    {
+        try
+        {
+            var processStartInfo = new ProcessStartInfo
+            {
+                FileName = gitApplication,
+                Arguments = $"{command} {arguments}",
+                RedirectStandardOutput = true,
+                UseShellExecute = false,
+                CreateNoWindow = true,
+                WorkingDirectory = repositoryDirectory ?? string.Empty,
+            };
+
+            using var process = Process.Start(processStartInfo);
+            if (process != null)
+            {
+                var output = process.StandardOutput.ReadToEnd();
+                process.WaitForExit();
```

Comment:
> process.WaitForExit(); [](http://example.com/codeflow?start=15&length=23)

add a timeout for this, otherwise you could be blocked indefinitely by external factors.

Reply:
Any reasonable timeout to allow for stuff that's potentially long-running (enormous GVFS monorepos?) will need to be quite long. Like, on the order of a minute or more. But I agree, that's probably better than just hanging the process and forcing a restart.

We'll have to reason a bit more about what if we can provide some experience to the user that "Hey, git.exe is taking forever! Maybe you have an orphaned index.lock file?". But that's a whole thing we'll need to dive into and refine as we get closer to public preview. I'll add this to our list of things to look into.

Reply:
Modified in latest commit. Thanks!